engine/src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -8,97 +8,73 @@ import cromwell.core.WorkflowId
-  case object AbortAllWorkflows extends WorkflowManagerActorMessage
+  case class SubmitWorkflowCommand(source: WorkflowSourceFiles) extends WorkflowManagerActorCommand
+  case class AbortWorkflowCommand(id: WorkflowId) extends WorkflowManagerActorCommand
+  case object AbortAllWorkflowsCommand extends WorkflowManagerActorCommand
+  private[WorkflowManagerActor] final case class RestartWorkflowsCommand(workflows: Seq[OldStyleWorkflowDescriptor]) extends WorkflowManagerActorCommand
 
-  def props(): Props = Props(new WorkflowManagerActor())
+  /**
+    * Responses
+    */
+  sealed trait WorkflowManagerActorResponse extends WorkflowManagerActorMessage
 
-  // FIXME hack to deal with one class of "singularity" where Cromwell isn't smart enough to launch only
-  // as much work as can reasonably be handled.
-  // How long to delay between restarting each workflow that needs to be restarted.  Attempting to
-  // restart 500 workflows at exactly the same time crushes the database connection pool.
-  lazy val RestartDelay = 200 milliseconds
+  def props(): Props = Props(new WorkflowManagerActor())
 
+  /**
+    * States
+    */
   sealed trait WorkflowManagerState
   case object Running extends WorkflowManagerState
   case object Aborting extends WorkflowManagerState
   case object Done extends WorkflowManagerState
 
-  case class WorkflowManagerData(workflows: Map[WorkflowId, WorkflowActorRef]) {
-    def add(entry: (WorkflowId, WorkflowActorRef)): WorkflowManagerData = this.copy(workflows = workflows + entry)
-    def remove(id: WorkflowId): WorkflowManagerData = this.copy(workflows = workflows - id)
-    def remove(actor: WorkflowActorRef): WorkflowManagerData = {
-      val workflowId = workflows.collectFirst { case (id, a) if a == actor => id }
+  /**
+    * Data
+    */
+  case class WorkflowManagerData(workflows: Map[WorkflowId, ActorRef]) {
+    def idFromActor(actor: ActorRef): Option[WorkflowId] =
+      workflows.collectFirst { case (id, a) if a == actor => id }
+    def withAddition(entry: WorkflowIdToActorRef): WorkflowManagerData =
+      this.copy(workflows = workflows + (entry.workflowId -> entry.workflowActor))
+    def without(id: WorkflowId): WorkflowManagerData =
+      this.copy(workflows = workflows - id)
+    def without(actor: ActorRef): WorkflowManagerData = {
       // If the ID was found in the lookup return a modified copy of the state data, otherwise just return
       // the same state data.
-      workflowId map remove getOrElse this
+      idFromActor(actor) map without getOrElse this
     }
   }
-
-  lazy val defaultConfig = ConfigFactory.load
 }
 
-/**
- * Responses to messages:
- * SubmitWorkflow: Returns a `Future[WorkflowId]`
- * WorkflowStatus: Returns a `Future[Option[WorkflowState]]`
- * WorkflowOutputs: Returns a `Future[Option[binding.WorkflowOutputs]]` aka `Future[Option[Map[String, WdlValue]]`
- *
- */
 class WorkflowManagerActor(config: Config)
```

Comment:
Are all the changes in this class also related to the de-shadowification (= enlightenment ?) 
![](http://www.islamiclife.com/2014/10/03/4413.jpg)